### PR TITLE
otlpjsonfilereceiver: add a replay_file config option to support heartbeats

### PR DESCRIPTION
**Pull request by ridwanmsharif**
**Description:** <Describe what has changed.>

This change adds a config option to allow users to replay static telemetry at an interval (poll_interval is the configurable interval).

This is useful for some  usecases like:
- Sending a heartbeat log to a logging backend, as a signal that the logging agent is alive
- Sending static metadata as metrics to metrics backend. The metadata can be read using the `otlpjsonfilereceiver` that uses this package and config.

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31533

**Testing:**
- Manual testing
- unit test for the fileconsumer
- test for the `otlpjsonfilereceiver`

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by atoulme** [comment]
Please add a changelog. Do you mean to use the PollInterval setting to set the replay interval?

**Comment by ridwanmsharif** [comment]
> Please add a changelog. Do you mean to use the PollInterval setting to set the replay interval?

Yes, `PollInterval` is what you would use to configure the intervals for the replay

File: pkg/stanza/fileconsumer/config.go
Comment (djaglowski) [inline pkg/stanza/fileconsumer/config.go:83]: I'd like to keep this flag out of the user facing API for the package, since it's not clear that it's broadly applicable.

What I'd like to do is put an options pattern in place, which will help with several other changes I anticipate for this package. Give me a day or two to work out what this looks like and then we can rebase to make this an option.

Comment (djaglowski) [reply pkg/stanza/fileconsumer/config.go:83]: I opened https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31596 to establish an options pattern. Once that is merged, let's add a `WithNoTracking` option. (For context on the suggested name, see [#31506](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31506#pullrequestreview-1917375290).) Then we can just pass in this option from only otlpjson receiver, [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f27143fae73a4a0631db8c55e46f6821d6121f76/receiver/otlpjsonfilereceiver/file.go#L76).

Comment (ridwanmsharif) [reply pkg/stanza/fileconsumer/config.go:83]: I agree with this pattern here, I think this replay option is only useful for the `otlpjsonfilereceiver`. Thanks for being so quick

Comment (djaglowski) [reply pkg/stanza/fileconsumer/config.go:83]: #31596 has been merged

Comment (ridwanmsharif) [reply pkg/stanza/fileconsumer/config.go]: d87a14c6ab4c7c80aa41b04c50a1bcb219315995 makes it so that when `noTracking` is set using the option, no known or previous files are tracked. Wasn't sure if we should wait for #31506 to land so the manager doesn't have to make these `!noTracking` checks everywhere. But I'm fine either way

Comment (djaglowski) [reply pkg/stanza/fileconsumer/config.go]: Good point. Might be easier overall to wait for #31506

Comment (ridwanmsharif) [reply pkg/stanza/fileconsumer/config.go]: Sweet, I'll update this PR today or tomorrow!

Comment (ridwanmsharif) [reply pkg/stanza/fileconsumer/config.go]: I lied earlier, i got distracted instead. I did update it now, PTAL

**Review by djaglowski — COMMENTED** [review]
Nice, I like how this came out. It gives me hope we can move further in the direction of modularizing various aspects of the fileconsumer package.

**Comment by djaglowski** [comment]
Code looks good to me generally, just some CI failures to address

**Review by djaglowski — APPROVED** [review]
Thanks @ridwanmsharif!

**Pull request merged**
